airflow/utils/log/s3_task_handler.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import warnings
+
+from airflow import configuration as conf
+from airflow.utils import logging as logging_utils
+from airflow.utils.log.file_task_handler import FileTaskHandler
+
+
+class S3TaskHandler(FileTaskHandler):
+    """
+    S3TaskHandler is a python log handler that handles and reads
+    task instance logs. It extends airflow FileTaskHandler and
+    uploads to and reads from S3 remote storage.
+    """
+
+    def __init__(self, base_log_folder, s3_log_folder):
+        super(S3TaskHandler, self).__init__(base_log_folder)
+        self.remote_base = s3_log_folder
+        self.log_relative_path = ''
+        # deprecated as of March 2016
+        if not self.remote_base and conf.get('core', 'S3_LOG_FOLDER'):
+            warnings.warn(
```

Comment:
As we are refactoring and this has been in for sometime should we just error out?